### PR TITLE
Km eth contract

### PR DESCRIPTION
**Pull request by thirtyaughtsix**

### CODE REVIEW

File: src/functionModules/components/FileListerModule.ts [hunk 1]
```diff
@@ -11,37 +13,36 @@ import { FlashType } from '../../models/Flash'
...
-                           newListingModule: NewListingModule,
-                           uploadModule: UploadModule) {
+  public static async list(appStore: Store<any>) {
 
+    const newListingModule = getModule(NewListingModule, appStore)
     try {
       newListingModule.setPercentComplete(50)
 
       // TODO: validate the listing?
       await ListingModule.postListing(
```
Comment (MelvinMallari) [inline src/functionModules/components/FileListerModule.ts]: Nit: `ListingModule` -> `ListingContractModule` I think may be more semantic (not to be confused with vuexModules)

Comment (thirtyaughtsix) [reply src/functionModules/components/FileListerModule.ts]: not a nit. needs to be replaced.  keep catching them all like pokemon.

File: src/functionModules/metamask/MetamaskModule.ts [hunk 2]
```diff
@@ -41,29 +49,54 @@ export default class MetamaskModule {
...
+      {
+        method: 'eth_sendTransaction',
+        params: [unsignedTransaction], // NOTE: do not miss that this is an array of 1
+        from: ethereum.selectedAddress,
       },
       (err: any, res: any) => {
         if (err) {
+          const flashesModule  = getModule(FlashesModule, appStore)
           flashesModule.append(new Flash(err, FlashType.error))
         } else if (successCallback) {
```
Comment (MelvinMallari) [inline src/functionModules/metamask/MetamaskModule.ts]: can just be `else`

Comment (MelvinMallari) [inline src/functionModules/metamask/MetamaskModule.ts]: Up to preference, but possibly more semantic option would be to pull out the input object as a variable, and callback as a function so we just have `ethereum.sendAsync(object, callback)`

File: src/functionModules/protocol/EtherTokenContractModule.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  public static async decreaseApproval(
+    account: string,
+    contractAddress: string,
+    amount: number,
+    appStore: Store<any>,
+    success: (
+      response: any,
+      appStore: Store<any>) => void,
+    transactOpts: TransactOpts) {
```
Comment (MelvinMallari) [inline src/functionModules/protocol/EtherTokenContractModule.ts]: nit: tab missing here

File: src/functionModules/protocol/EtherTokenContractModule.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  public static async deposit(
+    account: string,
+    amount: number,
+    appStore: Store<any>,
+    success: (
+      response: any,
+      appStore: Store<any>) => void,
+    transactOpts: TransactOpts) {
```
Comment (MelvinMallari) [inline src/functionModules/protocol/EtherTokenContractModule.ts]: nit: tab missing here

File: src/functionModules/protocol/EtherTokenContractModule.ts (hunk 3 above)
Comment (thirtyaughtsix) [reply src/functionModules/protocol/EtherTokenContractModule.ts]: these are correctly lined up

File: src/functionModules/metamask/MetamaskModule.ts (hunk 2 above)
Comment (thirtyaughtsix) [reply src/functionModules/metamask/MetamaskModule.ts]: this block is still tentative (for me the giveaway is the `err` and `res` param names).  let's leave it for now as the pattern is still in flux and is likely to change when the error handling gets implemented.

Comment (thirtyaughtsix) [reply src/functionModules/metamask/MetamaskModule.ts]: See above.

**Pull request merged**